components/wifi/esp8266-driver/ESP8266/ESP8266.cpp

```diff
@@ -772,8 +791,22 @@ void ESP8266::_oob_packet_hdlr()
             }
         }
         return;
-    } else if (!_parser.scanf("%d:", &amount)) {
-        return;
+    } else {
+        if (cip_info == 1) {
+            if (!_parser.scanf("%d,", &amount)) {
+                return;
+            }
+            if (!_parser.scanf("%15[^,],", _ip_buffer)) {
+                return;
+            }
+            if (!_parser.scanf("%d:", &port)) {
+                return;
+            }
+        } else {
+            if (!_parser.scanf("%d:", &amount)) {
+                return;
+            }
+        }
```

Comment:
Seems that `_parser.scanf("%d,", &amount)` is done whether `cip_info` is `1` or `0`, so let's just do it before the check?
What comes next could be merged so it takes less space and matches the style of how we normally do it in this file. So something like this:
```
        } else {
            if (!_parser.scanf("%d,", &amount)) {
                 return;
            }
            if (cip_info == 1) {
                if (!(_parser.scanf("%15[^,],", _ip_buffer) 
                    && _parser.scanf("%d:", &port)) {
                    return;
                }
           }
```
It takes less space, is equally readable and matches the way we normally do this in this file.

Reply:
Unfortunately you're wrong. 
Delimiter for "amount" is different in version with showing ip, port against to version without showing ip, port. In first case it is "," in second case is ":"

Reply:
True, I missed that.
so let's just merge the scanfs together:
```
if (!(_parser.scanf("%d,", &amount)
    && _parser.scanf("%15[^,],", _ip_buffer) 
    && _parser.scanf("%d:", &port)) {
    return;
}
```

Reply:
Done

Reply:
the current structure `if (!sth && !sthelse && !sthelse2)` will return if ALL conditions fail, so it will have to check all of them, which makes no sense if the first one failed. 
I think it is better to do `if (!(sth && sthelse && sthelse2))`, because then if one condition fails (for example the `%d,` match fails), the other conditions will not be checked ([short-circuit evaluation](https://en.wikipedia.org/wiki/Short-circuit_evaluation)).

Reply:
Done. Yes you're right. I've broken logic operation.